crates/ra_hir/src/semantics.rs

```diff
@@ -83,6 +83,11 @@ impl PathResolution {
 /// Primary API to get semantic information, like types, from syntax trees.
 pub struct Semantics<'db, DB> {
     pub db: &'db DB,
+    impl_: SemanticsImpl<'db>,
+}
+
+pub struct SemanticsImpl<'db> {
```

Comment:
Does it need to be `pub`?

Reply:
I'm not sure. It's now used in the `ToDef` trait and I changed that to take `SemanticsImpl`.

Reply:
Ah, right, than let's keep it pub for simplicity